config/initializers/assets.rb

```diff
@@ -8,5 +8,32 @@
 # Rails.application.config.assets.paths << Emoji.images_path
 
 # Precompile additional assets.
-# application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
-# Rails.application.config.assets.precompile += %w( search.js )
+# application.js, application.css, and all non-JS/CSS in the app/assets
+# folder are already added.
+
+# Rails.application.config.assets.precompile += %w( admin.js admin.css )
+class AssetsOptionsResponder
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    if env["REQUEST_METHOD"] == "OPTIONS"
+      [204, access_control_headers, []]
+    else
+      @app.call(env)
+    end
+  end
+
+  private
+
+  def access_control_headers
+    Rails.configuration.public_file_server.headers
+  end
+end
+
+Rails.application.configure do
+  if defined?(ActionDispatch::Static)
+    config.middleware.insert_before ActionDispatch::Static, AssetsOptionsResponder
```

Comment:
We will need to handle `OPTIONS` requests by ourselves. This is probably marginally better than `font_assets` gem which also handles normal respond headers and we just setup in the public server.

Should we go back to `font_assets`? Or with custom code like the above?

btw the above code needs to respond only to requests served by public file server. I'm not sure how to check the path yet. In worst case I can do like `font_assets` and check for file extension. But I don't like this approach a lot.